File: Jukebox/Jukebox/Model1.vb
Imports System
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity
Imports System.Linq

Partial Public Class Model1
    Inherits DbContext

    Public Sub New()
        MyBase.New("name=JukeboxContext")
    End Sub

    Public Overridable Property C__MigrationHistory As DbSet(Of C__MigrationHistory)
    Public Overridable Property Playlists As DbSet(Of Playlist)
    Public Overridable Property Songs As DbSet(Of Song)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Entity(Of Playlist)() _
            .HasMany(Function(e) e.Songs) _
            .WithOptional(Function(e) e.Playlist) _
            .HasForeignKey(Function(e) e.Playlists_ID)
    End Sub
End Class

File: Jukebox/Jukebox/Model1.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;
using System.Linq;

namespace Jukebox
{
    public partial class Model1 : DbContext
    {
        public Model1()
            : base("name=JukeboxSettings")
        {
        }

        public virtual DbSet<C__MigrationHistory> C__MigrationHistory { get; set; }
        //public virtual DbSet<Playlist> Playlists { get; set; }
        public virtual DbSet<Song> Songs { get; set; }

        public System.Data.Entity.DbSet<Jukebox.Playlist> Playlists { get; set; }
        /*
protected override void OnModelCreating(DbModelBuilder modelBuilder)
{
modelBuilder.Entity<Playlist>()
.HasMany(e => e.Songs)
.WithOptional(e => e.Playlist)
.HasForeignKey(e => e.Playlists_ID);
}
*/
    }
}
